toolkit/partner-chains-cli/src/prepare_configuration/mod.rs

```diff
@@ -195,6 +196,12 @@ impl FromStr for Protocol {
 	}
 }
 
+impl Display for Protocol {
+	fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+		write!(f, "{}", ToString::to_string(self))
```

Comment:
`ToString` uses `Display`, so I suspect this can be the cause of stack overflow:
https://doc.rust-lang.org/std/string/trait.ToString.html